### PR TITLE
Axis labels should respect the theme (#142) and fix #96 whilst I'm at it.

### LINKED ISSUE #96: Axes text to show up in the test figure

### LINKED ISSUE #142: Axis titles are always black.
We don't use axis or plot titles in all widgets but when we do, they are always black.
Screenshot 2023-06-01 at 10 04 06

This should be updated to respect the theme. Follows on from #86 / #138.

### PR DESCRIPTION
**Pull request by samcunliffe**
Fixes #142 by reapplying the colour scheme **_before_** ``canvas.draw`` in ``NapariMPLWidget._draw`` (cheers @dstansby!).

Since fixing the colour for axis labels means I need to regenerate the reference scatter plots, I thought I might as well fix #96 by setting the `viewer.theme` for every test that is marked ``@pytest.mark.mpl_image_compare``. It's not a fix via a nice global stylesheet setting but... you can now read the axes on the baseline figures. (Which is maybe nice??)

### Testing:

* A _somewhat_ trustworthy TDD reproduction is added into `test_theme`.

### Notes for reviewers:

Please double-check for me that a) `main` really does have #142, and that b) this really does solve it. I have had some strangeness that I am currently blaming on `pip install ` without `-e`.

### CODE REVIEW

File: src/napari_matplotlib/base.py [hunk 1]
```diff
@@ -246,6 +247,8 @@ def _draw(self) -> None:
         ):
             self.draw()
         self.canvas.draw()
+        self.canvas.flush_events()
+        self.apply_napari_colorscheme(self.figure.gca())
```
Comment (dstansby) [inline src/napari_matplotlib/base.py:251]:
```suggestion
        self.apply_napari_colorscheme(self.figure.gca())
        self.canvas.draw()
```
Might work?

**Review by ruaridhg — APPROVED** [review]
The axis labels were black when running napari from `main`, but running from this branch meant that they were white in dark mode and black in light mode so yay!

**Comment by ruaridhg** [comment]
Works locally, but tests are failing on tox because the existing baseline images are using the default theme so adding the them causes a difference
image

**Comment by samcunliffe** [comment]
> Works locally, but tests are failing on tox because the existing baseline images are using the default theme so adding the them causes a difference. Can figure out why next week.

I was replacing the plots in `tests/baseline` but didn't see there are also some baseline plots in `tests/scatter/baseline`.

Do I need to get into the habit of running ``pytest`` with ``--mpl-generate-path`` pointing to two directories (I guess with a ``-k`` to separate the plots into the scatter subdir?

**Comment by ruaridhg** [comment]
> > Works locally, but tests are failing on tox because the existing baseline images are using the default theme so adding the them causes a difference. Can figure out why next week.
> 
> I was replacing the plots in `tests/baseline` but didn't see there are also some baseline plots in `tests/scatter/baseline`.
> 
> Do I need to get into the habit of running `pytest` with `--mpl-generate-path` pointing to two directories (I guess with a `-k` to separate the plots into the scatter subdir?

Ah yes, that was a recent change by David. I've been running `pytest` with `--mpl-generate-path` twice with the different directories, but yes pointing to them both in the same command would be more efficient

**Review by ruaridhg — APPROVED** [review]
LGTM

**Pull request merged**